### ISSUE TITLE
env0_environment (Data Source) does not work when just the name is passed

### ISSUE THREAD

**Issue by arunnair-7** [env0/terraform-provider-env0#357]
On trying your example at https://registry.terraform.io/providers/env0/env0/latest/docs/data-sources/environment it doesn't work if just the name is passed

```
data "env0_environment" "by_name" {
  name = "best-env"
}
```
This is the error I get

```
root@b665ec491243:/code/cloudscript/terraform-modules/env0-blueprints/env0-environment# terraform plan
╷
│ Error: Could not get Environment: 400 Bad Request: You may specify exactly one of the parameters {}
│
│   with data.env0_environment.by_name,
│   on main.tf line 1, in data "env0_environment" "by_name":
│    1: data "env0_environment" "by_name" {
│
╵
```

But it works correctly if only the id is passed.

**Comment by omry-hay** [comment]
@arunnair-7 Thanks for reporting this bug.
Can you please specify the version of the Terraform provider that you have used? 
We've just released a fix for a different bug that should fix this issue as well in [v1.0.2](https://github.com/env0/terraform-provider-env0/releases/tag/v1.0.2).
Can you please upgrade and see if it is reproduced?

Thanks

**Comment by arunnair-7** [comment]
Hi @omry-hay I had initially tested this with v1.0.1
I just upgraded to https://github.com/env0/terraform-provider-env0/releases/tag/v1.0.2 and tried it but I got the same behavior.

It fails on using just the name and works correctly when you use just the id.
